process/datadog_checks/process/process.py

```diff
@@ -236,38 +236,35 @@ def psutil_wrapper(self, process, method, accessors, try_sudo, *args, **kwargs):
 
         try:
             res = getattr(process, method)(*args, **kwargs)
+
+            # Try to get `num_fds` using try_sudo
+            if method == 'num_fds' and self.try_sudo:
```

Comment:
The original condition had `Platform.is_unix()` in the `if` as well, we should keep it the same.

Reply:
Also I think the previous code should all be in an `else` block that only runs if this `if` doesn't. We don't need any of the other logic if we are using `try_sudo`. That includes the `res = getattr(process, method)(*args, **kwargs)` line above as well, and I would say the entire `try` block.

Reply:
I took out the `Platform.is_unix()` part because earlier in https://github.com/DataDog/integrations-core/pull/10109/files#diff-7f5384f67203af6eaba2a5de51e4b549939d49ab174a214b1b690b43986ecad8R232, non unix and `num_fds` would get filtered already.